### PR TITLE
Backport a couple of utilities from grapl-aws-plugins to grapl-tests-common

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
#231 

### What changes does this PR make to Grapl? Why?
Backport a couple of utilities from grapl-aws-plugins to grapl-tests-common
because they'll probably be useful for other tests in the future

### How were these changes tested?
run-e2e-tests

### CODE REVIEW

File: src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        hostname = "grapl-engagement-edge" if use_docker_links else "localhost"
+        self.endpoint = f"http://{hostname}:8900"
+
+    def get_jwt(self) -> str:
+        resp = requests.post(
+            f"{self.endpoint}/login",
+            json={
+                # hardcoded when IS_LOCAL
+                "username": "grapluser",
+                "password": "2ae5ddfb1eeeed45d502bcfd0c7b8f962f24bf85328ba942f32a31c0229c295a",
```
Comment (colin-grapl) [inline src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py:23]: Isn't the password graplpassword locally?

File: src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        plugin_dict: Dict[str, str] = {}
+        for path in paths:
+            with open(path, "r") as f:
+                clean_path = str(path).split("model_plugins/")[-1]
+                contents = f.read()
+                if len(contents) == 0:
+                    # attempted hack: make __init__.py non empty
+                    # due to https://github.com/boto/botocore/pull/1328
+                    # otherwise the client just hangs
+                    assert clean_path.endswith("__init__.py")
```
Comment (colin-grapl) [inline src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py:46]: I feel like it wouldn't really be an error to have an empty file not called __init__.py ? Might be better to just debug log here if it isn't.

File: src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            raise ModelPluginDeployerException(f"{resp.status_code}: {resp.text}")
+        return resp
+
+    def _is_valid_deployable_file(self, file_path: str) -> bool:
+        if file_path.endswith(".pyc"):
+            return False
+        if file_path.endswith(".ipynb"):
+            return False
+        # TODO: should this just check for `.py`?
+        # would there ever be any other extension we'd want to upload?
```
Comment (colin-grapl) [inline src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py:66]: Maybe someone wants to upload a txt file or something to use as part of their lib? idk

**Review by colin-grapl — APPROVED** [review]

File: src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py (hunk 1 above)
Comment (wimax-grapl) [reply src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py:23]: it's peppered and sha'd
(i briefly toyed with doing the pepper/sha here, but after 10 minutes of the resulting sha not being what i expected, i just decided 'fuck it')

File: src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py (hunk 2 above)
Comment (wimax-grapl) [reply src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py:46]: legit

**Pull request merged**